python/lsst/obs/base/script/convert.py

```diff
@@ -72,11 +71,20 @@ def convert(repo, gen2root, instrument, skymap_name, skymap_config, calibs, reru
         butlerConfig = repo
 
     butler = lsst.daf.butler.Butler(butlerConfig)
-    instr = getInstrument(instrument, butler.registry)
+
+    # Derive the gen3 instrument from the gen2root
+    # This requires we instantiate a gen2 butler solely to get its mapper
+    with temporaryLogLevel("CameraMapper", Log.ERROR):
+        with temporaryLogLevel("HscMapper", Log.ERROR):
+            butler2 = Butler2(gen2root)
+            gen2mapperClass = butler2.getMapperClass(gen2root)
+            del butler2
+
+    instrClass = gen2mapperClass.getGen3Instrument()
+    instr = instrClass()
```

Comment:
Any particular reason for abbreviating here?

Reply:
Intertia since that's what we called it before.